addons/l10n_latam_invoice_document/wizards/account_debit_note.py

```diff
@@ -1,16 +0,0 @@
-from odoo import models
-
-
-class AccountDebitNote(models.TransientModel):
-
-    _inherit = 'account.debit.note'
-
-    def create_debit(self):
-        """ Properly compute the latam document type of type debit note. """
-        res = super().create_debit()
-        new_move_id = res.get('res_id')
-        if new_move_id:
-            new_move = self.env['account.move'].browse(new_move_id)
-            new_move._compute_l10n_latam_document_type()
-            new_move._onchange_l10n_latam_document_type_id()
```

Comment:
Yes, would be nice to remove these hacks.

Reply:
Hi @jco-odoo, @jorenvo, and @vbe-odoo, I did some testing with this branch for EC localization, and I don't know if this change is strictly necessary. In the loom below I show how the document type does not show the correct EC type document when I create a debit note. Yes, I know that the allowed document type doesn't show other documents for debit notes, but on the first creation, it doesn't show the correct document type.

https://www.loom.com/share/7994e757ebd24635888ac162847c9f78?sid=276abc70-a3ae-4711-a1b7-8b2c4181a8d1

It might be good to stick with this code or in the `_prepare_default_values` method set the correct document type, to avoid this scenario:

https://github.com/odoo-dev/odoo/blob/a5afa7bb569179498161aa91fa0aba5c27d3d85b/addons/account_debit_note/wizard/account_debit_note.py#L50-L66